typing/typecore.ml

```diff
@@ -670,92 +674,109 @@ end) = struct
         in
         List.find check_type lbls
 
+  (** [disambiguate] selects a concrete description for [lid] using some
+      contextual information: an optional [expected_type], and a list of candidates [lbls].
```

Comment:
Because, when the list is empty, we may have an error message, which we may want to use later.

Reply:
If I understand the code correctly, the error comes from some label-filtering logic that was done by the caller, and we could get rid of this complication by passing the filter to the function instead.

Reply:
I'm afraid it would just increase the complication, as the caller would have to understand how this function is going to be called. But do we agree than any change in the API would have to be another PR anyway?

Reply:
I asked about how to proceed with further changes below. I agree with the idea of sending separate PRs, but I wished we would converge more quickly on each PR, because otherwise this could drag on indefinitely.

Concretely, would you be willing to approve the PR in its current state, even though we agree it's imperfect, so that follow-up PRs can start from `trunk` again? (I think we can iterate more on the documentation comment *and* I would like to experiment with the filtering logic, but those can go into separate PRs; the problem is that today they would conflict with the changes already in the PR.)